redaxo/src/addons/structure/functions/function_rex_url.php

```diff
@@ -57,10 +57,21 @@ function rex_getUrl($id = null, $clang = null, array $params = [], $separator =
 /**
  * Leitet auf einen anderen Artikel weiter.
  *
+ * @param int       $article_id
+ * @param null|int  $clang
+ * @param array     $params
+ *
+ * @throws InvalidArgumentException
+ *
  * @package redaxo\structure
  */
 function rex_redirect($article_id, $clang = null, array $params = [])
 {
+    // article id must be integer
+    if (!is_int($article_id)) {
```

Comment:
`null` und `''` sind auch nicht unüblich (für den aktuellen Artikel, zum Beispiel wenn man nur auf eine andere Sprache umleitet).
Auch ist die Frage, ob wir aus BC-Gründen auch Strings akzeptieren wollen, die reine Ints enthalten?  `'3'` etc.

Reply:
Ich würde auf Type int oder null setzen `''` oder `'1'` ist eher nicht Zielführend. Die Artikel ID ist ja definitiv Integer.

Reply:
Ein Beispiel, was danach nicht mehr funktionieren wird:

```php
rex_redirect(REX_LINK[id=1 output=id]);
```

Die Rex-Vars liefern immer Strings.

Reply:
Hier liegt der Fehler dann in `REX_LINK[id=1 output=id]`, es wird ja eine ID erwartet diese sollte auch den entsprechenden Typ haben.

Reply:
Will man eine Exception werfen aber den Type ignorieren sehe ich hier nur die Möglichkeit den Inhalt des Values zu prüfen `preg_match('/^\d+$/', $var)`.

Reply:
ich denke aber das diese Variante `if ((int)$var == $var) { return true; }`schneller wäre als ein matching.